### PR TITLE
Refactor rpcchainvm metrics registration

### LINKED ISSUE #3169: RPCChainVM Metrics Should Have Dedicated Namespace
Currently the `rpcchainvm` metrics are specified under the vm that they are executing.

For example:
```
avalanche_xsvm_rpcchainvm_grpc_server_msg_received_total{chain=a}
avalanche_subnetevm_rpcchainvm_grpc_server_msg_received_total{chain=b}
```

While these are guaranteed to be the exact same metrics, they are partitioned by vm (either `xsvm` or `subnetevm` here). The avalanchego side of the `rpcchainvm` metrics should be grouped together as:
```
avalanche_rpcchainvm_grpc_server_msg_received_total{chain=a}
avalanche_rpcchainvm_grpc_server_msg_received_total{chain=b}
```

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

Resolves #3169.

## How this works

1. Creates a metrics namespace for the avalanchego side of the rpcchainvm.
2. Removes the `plugin` prefix of metrics served over the grpc interface.

## How this was tested

- [X] CI
- [X] Locally running an `xsvm` subnet:

```
# HELP avalanche_rpcchainvm_bytes_to_id_cache_len number of entries
# TYPE avalanche_rpcchainvm_bytes_to_id_cache_len gauge
avalanche_rpcchainvm_bytes_to_id_cache_len{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_bytes_to_id_cache_portion_filled fraction of cache filled
# TYPE avalanche_rpcchainvm_bytes_to_id_cache_portion_filled gauge
avalanche_rpcchainvm_bytes_to_id_cache_portion_filled{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_bytes_to_id_cache_put_count number of put calls
# TYPE avalanche_rpcchainvm_bytes_to_id_cache_put_count counter
avalanche_rpcchainvm_bytes_to_id_cache_put_count{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_bytes_to_id_cache_put_time time spent (ns) in put calls
# TYPE avalanche_rpcchainvm_bytes_to_id_cache_put_time gauge
avalanche_rpcchainvm_bytes_to_id_cache_put_time{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_decided_cache_get_count number of get calls
# TYPE avalanche_rpcchainvm_decided_cache_get_count counter
avalanche_rpcchainvm_decided_cache_get_count{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",result="hit"} 166
# HELP avalanche_rpcchainvm_decided_cache_get_time time spent (ns) in get calls
# TYPE avalanche_rpcchainvm_decided_cache_get_time gauge
avalanche_rpcchainvm_decided_cache_get_time{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",result="hit"} 229964
# HELP avalanche_rpcchainvm_decided_cache_len number of entries
# TYPE avalanche_rpcchainvm_decided_cache_len gauge
avalanche_rpcchainvm_decided_cache_len{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 1
# HELP avalanche_rpcchainvm_decided_cache_portion_filled fraction of cache filled
# TYPE avalanche_rpcchainvm_decided_cache_portion_filled gauge
avalanche_rpcchainvm_decided_cache_portion_filled{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 1.519918441772461e-06
# HELP avalanche_rpcchainvm_decided_cache_put_count number of put calls
# TYPE avalanche_rpcchainvm_decided_cache_put_count counter
avalanche_rpcchainvm_decided_cache_put_count{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 3
# HELP avalanche_rpcchainvm_decided_cache_put_time time spent (ns) in put calls
# TYPE avalanche_rpcchainvm_decided_cache_put_time gauge
avalanche_rpcchainvm_decided_cache_put_time{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 2083
# HELP avalanche_rpcchainvm_grpc_server_handled_total Total number of RPCs completed on the server, regardless of success or failure.
# TYPE avalanche_rpcchainvm_grpc_server_handled_total counter
avalanche_rpcchainvm_grpc_server_handled_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_code="OK",grpc_method="Get",grpc_service="rpcdb.Database",grpc_type="unary"} 2
avalanche_rpcchainvm_grpc_server_handled_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_code="OK",grpc_method="Has",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_rpcchainvm_grpc_server_handled_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_code="OK",grpc_method="HealthCheck",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_rpcchainvm_grpc_server_handled_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_code="OK",grpc_method="WriteBatch",grpc_service="rpcdb.Database",grpc_type="unary"} 1
# HELP avalanche_rpcchainvm_grpc_server_msg_received_total Total number of RPC stream messages received on the server.
# TYPE avalanche_rpcchainvm_grpc_server_msg_received_total counter
avalanche_rpcchainvm_grpc_server_msg_received_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Get",grpc_service="rpcdb.Database",grpc_type="unary"} 2
avalanche_rpcchainvm_grpc_server_msg_received_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Has",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_rpcchainvm_grpc_server_msg_received_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="HealthCheck",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_rpcchainvm_grpc_server_msg_received_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="WriteBatch",grpc_service="rpcdb.Database",grpc_type="unary"} 1
# HELP avalanche_rpcchainvm_grpc_server_msg_sent_total Total number of gRPC stream messages sent by the server.
# TYPE avalanche_rpcchainvm_grpc_server_msg_sent_total counter
avalanche_rpcchainvm_grpc_server_msg_sent_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Get",grpc_service="rpcdb.Database",grpc_type="unary"} 2
avalanche_rpcchainvm_grpc_server_msg_sent_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Has",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_rpcchainvm_grpc_server_msg_sent_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="HealthCheck",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_rpcchainvm_grpc_server_msg_sent_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="WriteBatch",grpc_service="rpcdb.Database",grpc_type="unary"} 1
# HELP avalanche_rpcchainvm_grpc_server_started_total Total number of RPCs started on the server.
# TYPE avalanche_rpcchainvm_grpc_server_started_total counter
avalanche_rpcchainvm_grpc_server_started_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Get",grpc_service="rpcdb.Database",grpc_type="unary"} 2
avalanche_rpcchainvm_grpc_server_started_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Has",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_rpcchainvm_grpc_server_started_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="HealthCheck",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_rpcchainvm_grpc_server_started_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="WriteBatch",grpc_service="rpcdb.Database",grpc_type="unary"} 1
# HELP avalanche_rpcchainvm_missing_cache_len number of entries
# TYPE avalanche_rpcchainvm_missing_cache_len gauge
avalanche_rpcchainvm_missing_cache_len{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_missing_cache_portion_filled fraction of cache filled
# TYPE avalanche_rpcchainvm_missing_cache_portion_filled gauge
avalanche_rpcchainvm_missing_cache_portion_filled{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_missing_cache_put_count number of put calls
# TYPE avalanche_rpcchainvm_missing_cache_put_count counter
avalanche_rpcchainvm_missing_cache_put_count{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_missing_cache_put_time time spent (ns) in put calls
# TYPE avalanche_rpcchainvm_missing_cache_put_time gauge
avalanche_rpcchainvm_missing_cache_put_time{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_unverified_cache_len number of entries
# TYPE avalanche_rpcchainvm_unverified_cache_len gauge
avalanche_rpcchainvm_unverified_cache_len{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_unverified_cache_portion_filled fraction of cache filled
# TYPE avalanche_rpcchainvm_unverified_cache_portion_filled gauge
avalanche_rpcchainvm_unverified_cache_portion_filled{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_unverified_cache_put_count number of put calls
# TYPE avalanche_rpcchainvm_unverified_cache_put_count counter
avalanche_rpcchainvm_unverified_cache_put_count{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_rpcchainvm_unverified_cache_put_time time spent (ns) in put calls
# TYPE avalanche_rpcchainvm_unverified_cache_put_time gauge
avalanche_rpcchainvm_unverified_cache_put_time{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_xsvm_go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE avalanche_xsvm_go_gc_duration_seconds summary
avalanche_xsvm_go_gc_duration_seconds{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",quantile="0"} 1.1542e-05
avalanche_xsvm_go_gc_duration_seconds{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",quantile="0.25"} 2.9124e-05
avalanche_xsvm_go_gc_duration_seconds{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",quantile="0.5"} 3.4584e-05
avalanche_xsvm_go_gc_duration_seconds{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",quantile="0.75"} 3.7583e-05
avalanche_xsvm_go_gc_duration_seconds{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",quantile="1"} 4.275e-05
avalanche_xsvm_go_gc_duration_seconds_sum{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0.000249625
avalanche_xsvm_go_gc_duration_seconds_count{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 8
# HELP avalanche_xsvm_go_goroutines Number of goroutines that currently exist.
# TYPE avalanche_xsvm_go_goroutines gauge
avalanche_xsvm_go_goroutines{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 31
# HELP avalanche_xsvm_go_info Information about the Go environment.
# TYPE avalanche_xsvm_go_info gauge
avalanche_xsvm_go_info{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",version="go1.22.2"} 1
# HELP avalanche_xsvm_go_memstats_alloc_bytes Number of bytes allocated and still in use.
# TYPE avalanche_xsvm_go_memstats_alloc_bytes gauge
avalanche_xsvm_go_memstats_alloc_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 3.0150248e+07
# HELP avalanche_xsvm_go_memstats_alloc_bytes_total Total number of bytes allocated, even if freed.
# TYPE avalanche_xsvm_go_memstats_alloc_bytes_total counter
avalanche_xsvm_go_memstats_alloc_bytes_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 5.7294328e+07
# HELP avalanche_xsvm_go_memstats_buck_hash_sys_bytes Number of bytes used by the profiling bucket hash table.
# TYPE avalanche_xsvm_go_memstats_buck_hash_sys_bytes gauge
avalanche_xsvm_go_memstats_buck_hash_sys_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 1.464639e+06
# HELP avalanche_xsvm_go_memstats_frees_total Total number of frees.
# TYPE avalanche_xsvm_go_memstats_frees_total counter
avalanche_xsvm_go_memstats_frees_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 182327
# HELP avalanche_xsvm_go_memstats_gc_sys_bytes Number of bytes used for garbage collection system metadata.
# TYPE avalanche_xsvm_go_memstats_gc_sys_bytes gauge
avalanche_xsvm_go_memstats_gc_sys_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 3.38112e+06
# HELP avalanche_xsvm_go_memstats_heap_alloc_bytes Number of heap bytes allocated and still in use.
# TYPE avalanche_xsvm_go_memstats_heap_alloc_bytes gauge
avalanche_xsvm_go_memstats_heap_alloc_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 3.0150248e+07
# HELP avalanche_xsvm_go_memstats_heap_idle_bytes Number of heap bytes waiting to be used.
# TYPE avalanche_xsvm_go_memstats_heap_idle_bytes gauge
avalanche_xsvm_go_memstats_heap_idle_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 1.6113664e+07
# HELP avalanche_xsvm_go_memstats_heap_inuse_bytes Number of heap bytes that are in use.
# TYPE avalanche_xsvm_go_memstats_heap_inuse_bytes gauge
avalanche_xsvm_go_memstats_heap_inuse_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 3.3169408e+07
# HELP avalanche_xsvm_go_memstats_heap_objects Number of allocated objects.
# TYPE avalanche_xsvm_go_memstats_heap_objects gauge
avalanche_xsvm_go_memstats_heap_objects{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 78490
# HELP avalanche_xsvm_go_memstats_heap_released_bytes Number of heap bytes released to OS.
# TYPE avalanche_xsvm_go_memstats_heap_released_bytes gauge
avalanche_xsvm_go_memstats_heap_released_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 1.2853248e+07
# HELP avalanche_xsvm_go_memstats_heap_sys_bytes Number of heap bytes obtained from system.
# TYPE avalanche_xsvm_go_memstats_heap_sys_bytes gauge
avalanche_xsvm_go_memstats_heap_sys_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 4.9283072e+07
# HELP avalanche_xsvm_go_memstats_last_gc_time_seconds Number of seconds since 1970 of last garbage collection.
# TYPE avalanche_xsvm_go_memstats_last_gc_time_seconds gauge
avalanche_xsvm_go_memstats_last_gc_time_seconds{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 1.719979656002572e+09
# HELP avalanche_xsvm_go_memstats_lookups_total Total number of pointer lookups.
# TYPE avalanche_xsvm_go_memstats_lookups_total counter
avalanche_xsvm_go_memstats_lookups_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 0
# HELP avalanche_xsvm_go_memstats_mallocs_total Total number of mallocs.
# TYPE avalanche_xsvm_go_memstats_mallocs_total counter
avalanche_xsvm_go_memstats_mallocs_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 260817
# HELP avalanche_xsvm_go_memstats_mcache_inuse_bytes Number of bytes in use by mcache structures.
# TYPE avalanche_xsvm_go_memstats_mcache_inuse_bytes gauge
avalanche_xsvm_go_memstats_mcache_inuse_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 14400
# HELP avalanche_xsvm_go_memstats_mcache_sys_bytes Number of bytes used for mcache structures obtained from system.
# TYPE avalanche_xsvm_go_memstats_mcache_sys_bytes gauge
avalanche_xsvm_go_memstats_mcache_sys_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 15600
# HELP avalanche_xsvm_go_memstats_mspan_inuse_bytes Number of bytes in use by mspan structures.
# TYPE avalanche_xsvm_go_memstats_mspan_inuse_bytes gauge
avalanche_xsvm_go_memstats_mspan_inuse_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 257600
# HELP avalanche_xsvm_go_memstats_mspan_sys_bytes Number of bytes used for mspan structures obtained from system.
# TYPE avalanche_xsvm_go_memstats_mspan_sys_bytes gauge
avalanche_xsvm_go_memstats_mspan_sys_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 261120
# HELP avalanche_xsvm_go_memstats_next_gc_bytes Number of heap bytes when next garbage collection will take place.
# TYPE avalanche_xsvm_go_memstats_next_gc_bytes gauge
avalanche_xsvm_go_memstats_next_gc_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 5.219408e+07
# HELP avalanche_xsvm_go_memstats_other_sys_bytes Number of bytes used for other system allocations.
# TYPE avalanche_xsvm_go_memstats_other_sys_bytes gauge
avalanche_xsvm_go_memstats_other_sys_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 2.628441e+06
# HELP avalanche_xsvm_go_memstats_stack_inuse_bytes Number of bytes in use by the stack allocator.
# TYPE avalanche_xsvm_go_memstats_stack_inuse_bytes gauge
avalanche_xsvm_go_memstats_stack_inuse_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 1.048576e+06
# HELP avalanche_xsvm_go_memstats_stack_sys_bytes Number of bytes obtained from system for stack allocator.
# TYPE avalanche_xsvm_go_memstats_stack_sys_bytes gauge
avalanche_xsvm_go_memstats_stack_sys_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 1.048576e+06
# HELP avalanche_xsvm_go_memstats_sys_bytes Number of bytes obtained from system.
# TYPE avalanche_xsvm_go_memstats_sys_bytes gauge
avalanche_xsvm_go_memstats_sys_bytes{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 5.8082568e+07
# HELP avalanche_xsvm_go_threads Number of OS threads created.
# TYPE avalanche_xsvm_go_threads gauge
avalanche_xsvm_go_threads{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA"} 16
# HELP avalanche_xsvm_grpc_client_handled_total Total number of RPCs completed by the client, regardless of success or failure.
# TYPE avalanche_xsvm_grpc_client_handled_total counter
avalanche_xsvm_grpc_client_handled_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_code="OK",grpc_method="Get",grpc_service="rpcdb.Database",grpc_type="unary"} 2
avalanche_xsvm_grpc_client_handled_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_code="OK",grpc_method="Has",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_xsvm_grpc_client_handled_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_code="OK",grpc_method="HealthCheck",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_xsvm_grpc_client_handled_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_code="OK",grpc_method="WriteBatch",grpc_service="rpcdb.Database",grpc_type="unary"} 1
# HELP avalanche_xsvm_grpc_client_msg_sent_total Total number of gRPC stream messages sent by the client.
# TYPE avalanche_xsvm_grpc_client_msg_sent_total counter
avalanche_xsvm_grpc_client_msg_sent_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Get",grpc_service="rpcdb.Database",grpc_type="unary"} 2
avalanche_xsvm_grpc_client_msg_sent_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Has",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_xsvm_grpc_client_msg_sent_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="HealthCheck",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_xsvm_grpc_client_msg_sent_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="WriteBatch",grpc_service="rpcdb.Database",grpc_type="unary"} 1
# HELP avalanche_xsvm_grpc_client_started_total Total number of RPCs started on the client.
# TYPE avalanche_xsvm_grpc_client_started_total counter
avalanche_xsvm_grpc_client_started_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Get",grpc_service="rpcdb.Database",grpc_type="unary"} 2
avalanche_xsvm_grpc_client_started_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="Has",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_xsvm_grpc_client_started_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="HealthCheck",grpc_service="rpcdb.Database",grpc_type="unary"} 1
avalanche_xsvm_grpc_client_started_total{chain="2BoYBipXmQCEGexWusiMoAsoaHDJtmiwg1Q2BfbVjsDhvWEJaA",grpc_method="WriteBatch",grpc_service="rpcdb.Database",grpc_type="unary"} 1
```

### CODE REVIEW

File: vms/rpcchainvm/vm_client.go [hunk 1]
```diff
@@ -105,21 +106,23 @@ type VMClient struct {
...
 	}
 }
 
-// SetProcess gives ownership of the server process to the client.
-func (vm *VMClient) SetProcess(runtime runtime.Stopper, pid int, processTracker resource.ProcessTracker) {
-	vm.runtime = runtime
-	vm.processTracker = processTracker
-	vm.pid = pid
-	processTracker.TrackProcess(vm.pid)
-}
```
Comment (StephenButtolph) [inline vms/rpcchainvm/vm_client.go:121]: Technically getting rid of this function could be pulled into a separate PR... but it came up here because with both of these functions as is - it wasn't really clear where to provide the metrics gatherer.

**Review by aaronbuchwald — APPROVED** [review]
